yarn-project/acir-simulator/src/client/view_data_oracle.ts

```diff
@@ -35,6 +35,71 @@ export class ViewDataOracle extends TypedOracle {
     return this.db.getSecretKey(this.contractAddress, owner);
   }
 
+  /**
+   * Fetches the index and sibling path of a leaf at a given block from a given tree.
+   * @param blockNumber - The block number at which to get the membership witness.
+   * @param treeId - Id of the tree to get the sibling path from.
+   * @param leafValue - The leaf value
+   * @returns The index and sibling path concatenated [index, sibling_path]
+   */
+  public async getMembershipWitness(blockNumber: number, treeId: MerkleTreeId, leafValue: Fr): Promise<Fr[]> {
+    const index = await this.db.findLeafIndex(blockNumber, treeId, leafValue);
+    if (!index) {
+      throw new Error(`Leaf value: ${leafValue} not found in tree ${treeId}`);
+    }
+    const siblingPath = await this.db.getSiblingPath(blockNumber, treeId, index);
+    return [new Fr(index), ...siblingPath];
+  }
+
+  /**
+   * Fetches a sibling path at a given block and index from a tree specified by `treeId`.
+   * @param blockNumber - The block number at which to get the membership witness.
+   * @param treeId - Id of the tree to get the sibling path from.
+   * @param index - Index of the leaf to get sibling path for
+   * @returns The index and sibling path concatenated [index, sibling_path]
```

Comment:
incorrect return doc comment, there is no index

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/3255/commits/cea9ae82b6e5f6702f48b15116c1a38810e121f2